src/machines/modelingMachine.ts

```diff
@@ -715,8 +715,31 @@ export const modelingMachine = setup({
       if (event.type !== 'Revolve') return
       ;(async () => {
```

Comment:
Might be worthwhile taking the chance to move this to a fromPromise thing like other flows but it will likely make the diff less clean.

Reply:
I'll let you chime in on this @franknoirot @nadr0 when you have a chance (no rush)

Reply:
Eh might as well go for it